sqlakeyset/columns.py

```diff
@@ -126,7 +126,8 @@ def pair_for_comparison(self, value, dialect, apply_bind_processor: bool = False
             # If this OC is a column with a custom type, apply the custom
             # preprocessing to the comparison value:
             try:
-                value = compval.type.bind_processor(dialect)(value)  # type: ignore
+                type_impl = compval.type.dialect_impl(dialect)
+                value = type_impl.bind_processor(dialect)(value)
```

Comment:
Previously `compval.type` referred to `UtcDateTime.impl` or `DateTime` in the example I provided.
Now we resolve `compval.type` to the proper dialect specific implementation or `sqlite.DATETIME` in the example I provided.